grid2op/Environment/Environment.py

```diff
@@ -277,6 +277,7 @@ def _init_backend(
         self._check_rules_correct(legalActClass)
                 
         self._game_rules = RulesChecker(legalActClass=legalActClass)
+        self._game_rules.initialize(self)
```

Comment:
Something that comes to mind, when the env is copied (init_deepcopy_for_copy or something like this) do we need to do this ?

Reply:
Mirror question : is _game_rules class attribute is copied as a class or an instance with _custom_deepcopy_for_copy(self) ? Seems to be copy.deepcopy which supposedly copy each attribute of the gamerules, thus even the created ones when initialized. Can you confirm the thought ?

Reply:
Good question. the "deep copy" does not really work (yet) with grid2op.

The "_custom_deepcopy_for_copy" is here precisely because the deepcopy does not work. So I would not draw any conclusion from the behaviour of deepcopy to "_custom_deepcopy_for_copy". What does the implementation says about the `gamerules` ?